libraries/ESP8266WebServer/src/ESP8266WebServer.h

```diff
@@ -151,10 +151,15 @@ class ESP8266WebServerTemplate
 
   static String urlDecode(const String& text);
 
+  // send file based header and stream body when not "HTTP_HEAD" 
   template<typename T>
-  size_t streamFile(T &file, const String& contentType) {
+  size_t streamFile(T &file, const String& contentType, HTTPMethod requestMethod = HTTP_GET) {
     _streamFileCore(file.size(), file.name(), contentType);
-    return _currentClient.write(file);
+    if (requestMethod != HTTP_HEAD) {
+      return _currentClient.write(file);
+    } else {
+      return 0;
```

Comment:
Remove else and leave return statement as last line in the body

Reply:
done.